### PR TITLE
Add `error` and `exception` properties to Mailer messages

### PR DESCRIPTION
**Pull request by engram-design**
We've hit this a number of times in the past, but figured I should do something about it this time!

It's well-known that using the `Craft::$app->getMailer()->send($email)` function only returns a boolean for fail/success, which is a bit of a pain. Particularly in [Formie's](https://github.com/verbb/formie) case where we're smashed with issues on sending email notifications, commonly due to a misconfigured mailer, network issues, or anything related to specific mailers handling of emails.

In addition, we use [craft-log-to-file](https://github.com/putyourlightson/craft-log-to-file) to store all logs for the Formie plugin in a separate log file. This makes our support way, way more efficient.

I'm aware that Craft will save the error from the mailer in it's own `web.log` log, but this is tedious to get people to fish around in that for the error, when all other error information is included in our log file.

As email notifications are run in the queue, an error message is shown to CP users. At the moment, this can only be generic "Email failed", because we don't actively have the information as to _why_ the email failed to send. It'd be great if the error-handling you do:

https://github.com/craftcms/cms/blob/5bd4d4163f81156007bedc45138fe0f387089d6f/src/mail/Mailer.php#L174-L186

Could be stored somewhere, where plugins can access this, and report back to the user in a user-friendly way. I'm really just after the error message, but maybe not a bad idea to have the full exception there too, in case we want to display a full stack trace of the error.

Now we can use:

```
if (!Craft::$app->getMailer()->send($newEmail)) {
    Craft::dd($newEmail->error);
```

Long story short, please consider this 😄

Related to https://github.com/craftcms/cms/issues/7729

### CODE REVIEW

**Comment by brandonkelly** [comment]
Thanks for the PR! Changed the base to `3.7` since at this point 3.6 is just on maintenance mode, only receiving bug fixes.

Couple things:

- We should just set the actual exception to an `$error` property (rather than `$exception` to be consistent with `yii\queue\ExecEvent::$error`). The exception message can be fetched from `$message->error->getMessage()`.
- Need to check if `$message` is an instance of `craft\mail\Message` before assuming it actually has an `$error` property.

Also keep in mind that Swift Mailer [swallows some exceptions](https://github.com/swiftmailer/swiftmailer/blob/15f7faf8508e04471f666633addacf54c0ab5933/lib/classes/Swift/Mailer.php#L70-L76), which is why we haven’t tried to make this easier in the first place.

**Comment by engram-design** [comment]
Good call on the exception/error, a bit overkill with both. And yeah, annoying about those some of those errors, here's hoping it'll help a little. Updated!

**Comment by brandonkelly** [comment]
Merged for the next 3.7 release!

**Comment by brandonkelly** [comment]
3.7.0-beta.3 is out now with this change ✨
